packages/hierarchies/learning/HierarchyFiltering.md

```diff
@@ -236,3 +236,197 @@ The second step would be to create the node identifier paths. Let's consider two
+
+## Handling automatic expansion of nodes
+
+Imagine a case where we have this hierarchy:
+
+```txt
++ A
++--+ B
+|  +--+ C
+|     +--+ (...100 nodes)
++--+ ...
+```
+
+Let's say, we want the hierarchy to contain only the nodes with "C" label.
+In this case it may be useful to display to the user, where exactly "C" node is located without showing all of its 100 children.
+
+This is possible to do by providing an additional `autoExpand` option to the filter paths.
+All nodes in the filtered hierarchy up to "C" will have `autoExpand` flag on, so when rendering the hierarchy, it's clear when to stop loading the child nodes:
+
+```txt
++ A              (auto-expanded)
++--+ B           (auto-expanded)
+|  +--+ C        (collapsed)
+```
+
+The following code snippet shows, how to use `autoExpand` flag:
+
+<!-- [[include: [Presentation.Hierarchies.HierarchyFiltering.FilteringImports, Presentation.Hierarchies.HierarchyFiltering.AutoExpand], ts]] -->
+<!-- BEGIN EXTRACTION -->
+
+```ts
+import { createHierarchyProvider, HierarchyNodeIdentifiersPath } from "@itwin/presentation-hierarchies";
+import { ECSql, ECSqlQueryDef } from "@itwin/presentation-shared";
+
+// Construct a hierarchy provider for the filtered hierarchy
+const hierarchyProvider = createHierarchyProvider({
+  imodelAccess,
+  hierarchyDefinition: createHierarchyDefinition(imodelAccess),
+  filtering: {
+    paths: [
+      {
+        // Path to the element "C"
+        path: [elementKeys.a, elementKeys.b, elementKeys.c],
+        // Supply `autoExpand` flag with the path to the "C" element.
+        options: { autoExpand: true },
+      },
+    ],
+  },
+});
+
+// Collect the hierarchy & confirm we get what we expect - a hierarchy from root element "A" to target element "C"
+// Note that all nodes except C have `autoExpand` flag.
+expect(await collectHierarchy(hierarchyProvider)).to.containSubset([
+  {
+    label: "A",
+    autoExpand: true,
+    children: [
+      {
+        label: "B",
+        autoExpand: true,
+        children: [{ label: "C" }],
+      },
+    ],
+  },
+]);
+```
+
+<!-- END EXTRACTION -->
+
+Additionally, hierarchies may contain grouping nodes.
+Grouping nodes can be displayed to the user as any other nodes, however they do not represent actual instances, so filtering cannot be applied to them.
+Nevertheless, it's possible to "target" grouping node in the hierarchy by filtering the hierarchy by node's grouped instance keys.
+However in some cases it may be useful to leave this grouping node collapsed in the resulting hierarchy.
+
+For this purpose, there's an option to limit auto-expansion of nodes to a given grouping node that is expected to be in the resulting hierarchy.
+In order to do that, there has to be provided a minimal set of information about the grouping node: node's key and depth in the hierarchy, which can be determined from `parentKeys` attribute.
```

Comment:
```suggestion
In order to do that, there has to be provided a minimal set of information about the grouping node: node's key and depth in the hierarchy, which can be determined from the `parentKeys` attribute.
```

You say that we need to supply node's key and depth, but we call the latter `parentKeysCount`. Maybe the attribute should be renamed to `depth`? I think that would make it's meaning more obvious.

Reply:
I think this is a double-edged sword. On the other hand, `depth` is more clear in regards to what this means in the internal logic, on the other hand `parentKeysCount` is more clear for the consumer in regards to what needs to be passed there. I am not sure what's better.

Reply:
Will rename to `depth`